### PR TITLE
[WIP] PDBBind loader

### PR DESCRIPTION
**Pull request by ncfrey**
This PR  updates the PDBBind datasets in MolNet with `deepchem.molnet.load_function.pdbbind_datasets.load_pdbbind` function following the new API conventions. It includes v2019 of the "refined-set" of PDBBind. Details and references are provided in the docstring of `load_pdbbind`.

TODO:
* PDBBind uses interaction fingerprints which inherit from `ComplexFeaturizer`, which unfortunately does not inherit from `Featurizer`. For this reason, there are type errors because `_MolnetLoader` is expecting a subclass of `Featurizer`. Any ideas on how to fix these errors without substantial modifications to `_MolnetLoader` would be appreciated.
* If there is interest or compelling reasons to include legacy PDBBind datasets, I can add an argument so that the loader can access old datasets.

CCing @rbharath @peastman

### CODE REVIEW

**Comment by peastman** [comment]
Perhaps we should make ComplexFeaturizer be a subclass of Featurizer?  It looks like this was probably just done for historical reasons.  I only see two ways in which its API is inconsistent with Featurizer:

1. `featurize()` takes its input as two sequences, `mol_files: Sequence[str], protein_pdbs: Sequence[str]`, rather than a single one, `datapoints: Iterable[Any]`.  We could easily combine them into a single sequence of tuples.

2. `featurize()` returns a list of files that it couldn't featurize.  That's in contrast to other featurizers that just log errors to the console.  It's not clear to me why this featurizer should use a different error handling mechanism than all others?

File: deepchem/molnet/load_function/pdbbind_datasets.py [hunk 1]
```diff
@@ -1,458 +1,124 @@
...
-    data_dir = deepchem.utils.data_utils.get_data_dir()
-    if reload:
-      save_dir = os.path.join(
-          data_dir, "pdbbind_" + subset + "/" + featurizer + "/" + str(split))
-
-    dataset_file = os.path.join(data_dir, subset + "_smiles_labels.csv")
+class _PDBBindLoader(_MolnetLoader):
 
+  def create_dataset(self) -> Dataset:
+    dataset_file = os.path.join(self.data_dir, "pdbbind_v2019_refined.tar.gz")
```
Comment (rbharath) [inline deepchem/molnet/load_function/pdbbind_datasets.py:19]: Could we add options to get the full set as well? Might be useful for benchmarking methods at larger scale

File: deepchem/molnet/load_function/pdbbind_datasets.py [hunk 2]
```diff
@@ -1,458 +1,124 @@
...
+  The PDBBind dataset includes experimental binding affinity data
+  and structures for 4852 protein-ligand complexes from the "refined set"
+  in PDBBind v2019. The refined set removes data with obvious problems
+  in 3D structure, binding data, or other aspects and should therefore
+  be a better starting point for docking/scoring studies. Details on
+  the criteria used to construct the refined set can be found in [4]_.
 
-  # Extract locations of data
-  pdbs = []
+  Random splitting is recommended for this dataset.
```
Comment (rbharath) [inline deepchem/molnet/load_function/pdbbind_datasets.py:82]: This is out of scope for the current PR, but we might want to add a splitter based on sequence dissimilarity or potentially even an evolutionary splitter to measure generalizability across proteins in a future PR

**Review by rbharath — COMMENTED** [review]
Awesome, thanks for getting this off the ground @ncfrey! Some thoughts:

- I think it would be a great idea to make `ComplexFeaturizer` a subclass of `Featurizer`. It makes logical sense and would clean up the API nicely. 
- On the error handling, I think that's another historical artifact, perhaps induced by the fact that the PDB loaders had more errors than the molecular loaders (which fail rarely). I think we could swap this to have the same error handling mechanism as everything else.
- I think there isn't a strong reason to have legacy support retained. Very few (if any) people have been using the DeepChem protein-ligand support. As we're modernizing, we might as well upgrade to the latest here

**Comment by ncfrey** [comment]
Ok great, thanks for the feedback! I'll start a separate PR to update `ComplexFeaturizer`, since I think it will introduce some ripple effects that I'll have to work through.

**Pull request closed without merging**